sbi/utils/sbiutils.py

```diff
@@ -10,10 +10,37 @@
 from torch import nn as nn
 from torch import ones, zeros
 from tqdm.auto import tqdm
+import warnings
 
 from sbi.utils.torchutils import BoxUniform
 
 
+def warn_if_zscoring_changes_data(x: Tensor, duplicate_tolerance: float = 0.1) -> None:
+    """Raise warning if z-scoring would create duplicate data points.
+    
+    Args:
+        x: data
+        duplicate_tolerance: tolerated proportion of duplicates after z-scoring.
+    """
+
+    # Count unique xs.
+    num_unique = torch.unique(x, dim=0).numel()
+
+    # z-score.
+    zx = (x - x.mean(0)) / x.std(0)
+
+    # Count again and warn warn if too many new duplicates.
+    num_unique_z = torch.unique(zx, dim=0).numel()
+
+    if (1 - duplicate_tolerance) * num_unique_z < num_unique:
+        warnings.warn(
+            """Z-scoring these data will result in many data points being mapped on the
+            same z-scored value. This can occur due to numerical inaccuracies when the
+            data covers a large range of values. Consider setting z_score_x=False.""",
+            UserWarning,
+        )
```

Comment:
Will `z_score_x=False` really solve things? When there are such strong outliers in the `x`, they might also break NN training, no? How about suggesting the user should get rid of extreme outliers?

Reply:
I worked fine for my connectomics problem :) 
<img width="244" alt="image" src="https://user-images.githubusercontent.com/11921527/107032130-9a45ba80-67b3-11eb-8079-c811a35f8810.png">

Reply:
but you are right, we could add a sentence in the warning hinting at that.